### ISSUE TITLE
Cryptic cgroup v2 error message when starting rootfull containers

### ISSUE THREAD

**Issue by danopia** [containers/podman#8881]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I've installed Podman from the Kubic repo on a fresh Debian 10 / buster VM, which is provided by Chromebook's "Linux Beta" feature.

When I try running a container with `sudo`, I get a seemingly incorrect message

**Steps to reproduce the issue:**

1. Obtain Chromebook, enable Linux, install Podman (maybe there's a better way to repro, sorry...)

2. `sudo podman run --rm busybox echo 'hi world'`

**Describe the results you received:**

`Error: OCI runtime error: cgroup v2 not enabled on this host, can't use systemd (rootless) as cgroups manager`

**Describe the results you expected:**

`hi world`

**Additional information you deem important (e.g. issue happens only occasionally):**

This error is new to me; I have used previous podmans on this computer before wiping the Linux environment and they worked with `sudo`.

The error message's rootless mention seems incorrect as the command is being blatantly run with `sudo`.

To the best of my knowledge, this Linux has both cgroup v1 and v2 compiled in, while only v1 is mounted / in use.

**Output of `podman version`:**

```
Version:      2.2.1
API Version:  2.1.0
Go Version:   go1.14
Built:        Thu Jan  1 01:00:00 1970
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.18.0
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: 'conmon: /usr/libexec/podman/conmon'
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.22, commit: '
  cpus: 2
  distribution:
    distribution: debian
    version: "10"
  eventLogger: journald
  hostname: penguin
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.4.67-09356-gf3ed4c0c4a31
  linkmode: dynamic
  memFree: 15239045120
  memTotal: 15260381184
  ociRuntime:
    name: runc
    package: 'cri-o-runc: /usr/lib/cri-o-runc/sbin/runc'
    path: /usr/lib/cri-o-runc/sbin/runc
    version: 'runc version spec: 1.0.2-dev'
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  rootless: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 0
  swapTotal: 0
  uptime: 22h 33m 55s (Approximately 0.92 days)
registries:
  search:
  - docker.io
  - quay.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: btrfs
  graphOptions: {}
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Build Version: 'Btrfs v5.9 '
    Library Version: "102"
  imageStore:
    number: 1
  runRoot: /var/run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 2.1.0
  Built: 0
  BuiltTime: Thu Jan  1 01:00:00 1970
  GitCommit: ""
  GoVersion: go1.14
  OsArch: linux/amd64
  Version: 2.2.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
> apt list podman
Listing... Done
podman/unknown,now 2.2.1~4 amd64 [installed]
podman/unknown 2.2.1~4 arm64
podman/unknown 2.2.1~4 armhf
podman/unknown 2.2.1~4 ppc64el
podman/unknown 2.2.1~4 s390x
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

AFAIK, Chromebook "Linux Beta" is implemented using LXC containers.

**Comment by mheon** [comment]
After some grepping, this error looks like it's coming out of `runc`. You might want to open an issue against github.com/opencontainers/runc about this?

**Comment by danopia** [comment]
Ah, I suppose that's what the "OCI runtime error" prefix was trying to say.

I believe their code actually answers why the error message is showing up. They assume that rootful stuff shouldn't be done if we're already in a userns, falling back to rootless-as-root I guess. And I am in fact inside a userns because Chromebook Linux beta is implemented with LXC.

```
	// euid = 0, in a userns.
	// As we are unaware of cgroups path, we can't determine whether we have the full
	// access to the cgroups path.
	// Either way, we can safely decide to use the rootless cgroups manager.
```

Ok, I'll file with upstream, and see how they feel about this. It does seem like runc already has an escape hatch of [a 'rootless' bool](https://github.com/opencontainers/runc/blob/5efdc0c5a42d7965141271448dba6af4e1429962/rootless_linux.go#L14) which I assume can't be reached through podman as is?

**Comment by danopia** [comment]
I figured out what was causing my regression, kinda.

This binary that I don't know where it came from: (dpkg says it doesn't own it)
```
> ls -l /usr/lib/cri-o-runc/sbin/runc 
-rwxr-xr-x 1 root root 13441208 Dec  8 21:05 /usr/lib/cri-o-runc/sbin/runc*
> /usr/lib/cri-o-runc/sbin/runc --version
runc version spec: 1.0.2-dev
```

I removed it so now only the dpkg runc is left:
```
> runc --version
runc version 1.0.0~rc6+dfsg1
commit: 1.0.0~rc6+dfsg1-3
spec: 1.0.1
```

And now rootfull things work around.

```
> sudo podman run --rm busybox echo 'hi world'
hi world
```

Wild.

**Comment by mheon** [comment]
That is a very old `runc` so it makes sense that it would have issues. We used to ship a `runc` at that location, but that was the better part of two years ago.

**Comment by danopia** [comment]
It's actually the other way around on this setup — _older_ runc versions _don't_ have any obvious issue. Something snuck in a modern runc since I built this container a week ago and that broke me.

I just ran an automatic git bisect which is decisively showing that this runc commit is what breaks my podman: https://github.com/opencontainers/runc/commit/bf15cc99b1e7b486dfbb394992fcc4b38157d56e so it's starting to look like a pretty straightforward issue for runc to triage.

Apparently, due to LXC, I've always been running containers in a rootless-as-root mode. The more you know...

I'm going to go ahead and close this as the issue is proven to be runc.
